src/main/webapp/app/modules/conflicts/single/components/opening-hours-details.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import '../single-record-view.scss';
+import { connect } from 'react-redux';
+import InputField from './input-field';
+import { IActivity } from 'app/shared/model/activity.model';
+import { IOpeningHours } from 'app/shared/model/opening-hours.model';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { Translate } from 'react-jhipster';
+import { Collapse, Table } from 'reactstrap';
+
+export interface IOpeningHoursDetailsProp extends StateProps, DispatchProps {
+  activity: IActivity;
+  hours: IOpeningHours[];
+}
+
+export interface IOpeningHoursDetailsState {
+  isAreaOpen: boolean;
+}
+
+export class OpeningHoursDetails extends React.Component<IOpeningHoursDetailsProp, IOpeningHoursDetailsState> {
+  state: IOpeningHoursDetailsState = {
+    isAreaOpen: true
+  };
+
+  toggleAreaOpen = () => {
+    this.setState({
+      isAreaOpen: !this.state.isAreaOpen
+    });
+  };
+
+  mapWeekdayToString = day => {
+    switch (day) {
+      case 0:
+        return 'Monday';
+      case 1:
+        return 'Tuesday';
+      case 2:
+        return 'Wednesday';
+      case 3:
+        return 'Thursday';
+      case 4:
+        return 'Friday';
+      case 5:
+        return 'Saturday';
+      case 6:
+        return 'Sunday';
+      default:
+        return day;
+    }
+  };
+
+  render() {
+    const { hours } = this.props;
+    return (
+      <div>
+        <h4 className="title">
+          <div className="collapseBtn" onClick={this.toggleAreaOpen}>
+            <div className="collapseIcon">
+              <FontAwesomeIcon size="xs" icon={this.state.isAreaOpen ? 'angle-up' : 'angle-down'} />
+            </div>
+            <Translate contentKey="singleRecordView.details.openingHoursTitle" />
+          </div>
+        </h4>
```

Comment:
Can you take h4 to another component

Reply:
Sure, but I think it would increase the complexity needleesly, as it is responsible for hiding and displaying the whole section, and we would have to manage to change states through callbacks here.

Reply:
ok